src/logger/loggerApi.c

```diff
@@ -1088,22 +1091,35 @@ static const jsmntok_t * setPidExtendedField(const jsmntok_t *valueTok, const ch
 int api_setObd2Config(Serial *serial, const jsmntok_t *json){
 	OBD2Config *obd2Cfg = &(getWorkingLoggerConfig()->OBD2Configs);
 
-	setUnsignedCharValueIfExists(json, "en", &obd2Cfg->enabled, NULL);
-	size_t pidIndex = 0;
-	const jsmntok_t *pidsTok = findNode(json, "pids");
+	int pidIndex = 0;
+	setIntValueIfExists(json, "index", &pidIndex);
 
-	if (pidsTok != NULL && (++pidsTok)->type == JSMN_ARRAY) {
-		size_t arrSize = pidsTok->size;
+	if (pidIndex >= OBD2_CHANNELS){
+		return API_ERROR_PARAMETER;
+	}
 
-      for (pidsTok++; pidIndex < arrSize; pidIndex++) {
-         PidConfig *pidCfg = obd2Cfg->pids + pidIndex;
-         ChannelConfig *chCfg = &(pidCfg->cfg);
-         pidsTok = setChannelConfig(serial, pidsTok, chCfg, setPidExtendedField, pidCfg);
-      }
-   }
+	const jsmntok_t *pidsTok = findNode(json, "pids");
+	if (pidsTok != NULL && (++pidsTok)->type == JSMN_ARRAY) {
```

Comment:
[Feedback]: this is borderline too complex (with the increment inline in the second statement).  Normally I would use consts values and let the compiler handle the optimizations.